src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -210,10 +217,15 @@ public static void updateRepoConfig(RepoConfiguration config) {
     /**
      * Find and update {@code config} with all the author identities if author list is empty.
      */
-    private static void updateAuthorList(RepoConfiguration config) {
+    private static void updateAuthorList(RepoConfiguration config) throws NoAuthorsFoundWithCommitsException {
         if (config.getAuthorList().isEmpty()) {
             logger.info(String.format(MESSAGE_NO_AUTHORS_SPECIFIED, config.getLocation(), config.getBranch()));
             List<Author> authorList = GitShortlog.getAuthors(config);
+
+            if (authorList.isEmpty()) {
+                throw new NoAuthorsFoundWithCommitsException("Unable to find author with commits in this repository.");
```

Comment:
The message `Unable to find author ..` is unnecessary if the name of the exception already implies the error and you did not use the message.